### PR TITLE
Offer tutorial and usage examples in readme

### PR DESCRIPTION
**Pull request by thomashoneyman**
I've written a guide to using the `routing-duplex` library to serve as a README. It starts with a short summary and usage example, but then walks through several more examples in the style of `heterogeneous`.

There may be some typos or things you'd like to correct about how I've described the library and its use, and I welcome those! I'd love to have this up for others to be able to begin using the library.

### CODE REVIEW

**Comment by paulyoung** [comment]
Awesome! Will take a look later.

**Comment by thomashoneyman** [comment]
This should probably include guidance on hash-based routes, too.

File: README.md [hunk 1]
```diff
@@ -1 +1,419 @@
...
+```
+
+At this point, our codec is looking much cleaner:
+
+```purescript
+route = root $ sum
+  { "Root": noArgs
+  , "Profile": "user" / segment
+  , "Post": "user" / segment / "post" / int segment
+  , "Feed": "feed" / params { search: optional <<< string }
```
Comment (natefaubion) [inline README.md]: Like `string` is unnecessary with `segment`, it's also unnecessary here. Maybe we could introduce `?` here as well now that it works with strings

```purescript
  , "Feed": "feed" ? { search: optional }
```

File: README.md [hunk 2]
```diff
@@ -1 +1,419 @@
...
+* `/edit/:id` should represent updating
+
+Exactly the same way we wrote a codec for our `Route` type we can write one for our new `CRU` type:
+
+```purescript
+cru :: forall a. RouteDuplex' a -> RouteDuplex' (CRU a)
+cru inner = sum
+  { "Create": noArgs
+  , "Read": inner
+  , "Update": path "edit" inner
```
Comment (natefaubion) [inline README.md]: Should this use `"edit" / inner` syntax?

File: README.md (hunk 1 above)
Comment (thomashoneyman) [reply README.md]: 👍

File: README.md (hunk 2 above)
Comment (thomashoneyman) [reply README.md]: 👍

Comment (thomashoneyman) [reply README.md]: If `suffix` is ready, then I could change this to be `inner / suffix "edit"` or whatever correct usage is of that function -- that at least shows off the possibility.

**Comment by thomashoneyman** [comment]
Updated based on your notes.

**Pull request merged**